app/src/main/java/com/duckduckgo/app/browser/useragent/UserAgentRewriter.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.useragent
+
+import android.net.Uri
+import com.duckduckgo.app.global.UriString
+import com.duckduckgo.app.global.baseHost
+import com.duckduckgo.app.global.isMobileSite
+import javax.inject.Inject
+
+class MobileUrlReWriter @Inject constructor() {
+
+    fun isStrictlyMobileSite(uri: Uri?): Boolean {
```

Comment:
Reading this method is confusing me, but I'm having problems finding another name so I will try to explain why I was confused.

What I expected reading this method:
- isStrictlyMobileSite("m.twitter.com") -> true (but it's false)
- isStrictlyMobileSite("facebook.com") -> false (but it's true)
- isStrictlyMobileSite("m.facebook.com") -> true (but it's false)
- isStrictlyMobileSite("example.com") -> false (and it's false)

Using the previous example: isStrictlyMobileSite("facebook.com") -> returns true, but `true` means is that we need to change something and force the mobile site version. isStrictlyMobileSite("m.facebook.com") will return false, because it's already a mobile site.

Let me know if you want to chat about it.

Reply:
Changed to `mobileSiteOnlyForUri` which now returns `MobileSiteOnly?` instead of `true/false`. I think this makes more sense and allows us to get the mobile url directly from the object.